File: app/src/main/java/com/tregula/graedukacyjna/view/mode/ModeChooseView.java
package com.tregula.graedukacyjna.view.mode;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.Button;
import android.widget.LinearLayout;

import com.tregula.graedukacyjna.R;

import androidx.annotation.Nullable;

public class ModeChooseView extends LinearLayout {

    private Button learn_mode_button;
    private Button test_mode_button;

    public ModeChooseView(Context context) {
        super(context);
        init(context);
    }

    public ModeChooseView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public ModeChooseView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    public void init(Context context){
        inflate(context, R.layout.mode_choose_view,this);
        learn_mode_button=findViewById(R.id.learn_mode_button);
        test_mode_button=findViewById(R.id.test_mode_button);

    }
}

File: app/src/main/java/com/tregula/graedukacyjna/view/mode/ModeChooseView.kt
package com.tregula.graedukacyjna.view.mode

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.Button
import androidx.constraintlayout.widget.ConstraintLayout
import com.tregula.graedukacyjna.R

class ModeChooseView @JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : ConstraintLayout(context, attrs, defStyleAttr) {

    var onLearnClick: () -> Unit = {}
    var onTestClick: () -> Unit = {}

    init {
        View.inflate(context, R.layout.mode_choose_view, this)
        val learnMode: Button = findViewById(R.id.learn_mode_button)
        val testMode: Button = findViewById(R.id.test_mode_button)

        learnMode.setOnClickListener {
            onLearnClick()
        }
        testMode.setOnClickListener {
            onTestClick()
        }
    }
}
